### PR TITLE
`nix-env --query`: fix `--json` ignoring `--drv-path`

### PR DESCRIPTION
**Pull request by Artturin**
```json
{
  "AMB-plugins": {
    "drvPath": "/nix/store/l99cb7h2hy8dg005arsjbd9kx0w05d3h-AMB-plugins-0.8.1.drv",
    "name": "AMB-plugins-0.8.1",
    "outputName": "out",
    "outputs": {
      "out": null
    },
    "pname": "AMB-plugins",
    "system": "x86_64-linux",
    "version": "0.8.1"
  },
  "ArchiSteamFarm": {
    "drvPath": "/nix/store/nhplgyjj34fz6hjmnyih25gxscfh8s7b-ArchiSteamFarm-5.4.12.5.drv",
    "name": "ArchiSteamFarm-5.4.12.5",
    "outputName": "out",
    "outputs": {
      "out": null
    },
    "pname": "ArchiSteamFarm",
    "system": "x86_64-linux",
    "version": "5.4.12.5"
  },
...
```

# Motivation

# Context

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

File: src/nix-env/nix-env.cc [hunk 1]
```diff
@@ -953,6 +953,11 @@ static void queryJSON(Globals & globals, std::vector<DrvInfo> & elems, bool prin
                 }
             }
 
+            if (printDrvPath) {
+                auto drvPath = i.queryDrvPath();
+                if (drvPath) pkgObj["drvPath"] = globals.state->store->printStorePath(*drvPath);
```
Comment (Artturin) [inline src/nix-env/nix-env.cc:958]: For ref this is the code used in the non-json path

```
            if (printDrvPath) {
                auto drvPath = i.queryDrvPath();
                if (xmlOutput) {
                    if (drvPath) attrs["drvPath"] = store.printStorePath(*drvPath);
                } else
                    columns.push_back(drvPath ? store.printStorePath(*drvPath) : "-");
            }
```

**Comment by Ericson2314** [comment]
@Artturin Can you add a test (and maybe also release notes)?

https://github.com/NixOS/nix/pull/9242 also tests JSON which is non-deteriministic due to the store path, so it might be useful to look at that.

**Comment by Artturin** [comment]
> @Artturin Can you add a test (and maybe also release notes)?
> 
> #9242 also tests JSON which is non-deteriministic due to the store path, so it might be useful to look at that.

Done

File: doc/manual/src/release-notes/rl-next.md [hunk 2]
```diff
@@ -12,6 +12,8 @@
 
 - Introduce a new built-in function [`builtins.convertHash`](@docroot@/language/builtins.md#builtins-convertHash).
 
+- `nix-env --query` now respects `--drv-path` with `--json`.
```
Comment (fricklerhandwerk) [inline doc/manual/src/release-notes/rl-next.md:15]: I'm not sure what this is supposed to mean, could you please expand? The PR description provides no context.

Is this also documented in the manual?

**Review by fricklerhandwerk — DISMISSED** [review]

File: doc/manual/src/release-notes/rl-next.md (hunk 2 above)
Comment (Artturin) [reply doc/manual/src/release-notes/rl-next.md:15]: > I'm not sure what this is supposed to mean, could you please expand? The PR description provides no context.
> 
> Is this also documented in the manual?

It just means that drv-path works with --json

Comment (fricklerhandwerk) [reply doc/manual/src/release-notes/rl-next.md:15]:
```suggestion
- Fixed a bug where `nix-env --query` ignored `--drv-path` when `--json` was set.
```

**Review by Ericson2314 — APPROVED** [review]

**Comment by Artturin** [comment]
@fricklerhandwerk either approve or dismiss your review so the auto merge will work

**Pull request merged**